### PR TITLE
Soften shadows

### PR DESCRIPTION
**Pull request by yjbanov**
Make shadows softer by pre-multiplying the alpha by 0.4 (the number is picked purely empirically by comparing the output with Flutter for desktop).

Fixes https://github.com/flutter/flutter/issues/52734

This PR requires https://github.com/flutter/goldens/pull/84

### CODE REVIEW

File: lib/web_ui/test/golden_tests/engine/shadow_golden_test.dart [hunk 1]
```diff
@@ -12,7 +12,7 @@ import 'package:web_engine_tester/golden_tester.dart';
 
 import 'scuba.dart';
 
-const Color _kShadowColor = Color.fromARGB(255, 255, 0, 0);
```
Comment (ferhatb) [inline lib/web_ui/test/golden_tests/engine/shadow_golden_test.dart]: why the change in test?

Comment (yjbanov) [reply lib/web_ui/test/golden_tests/engine/shadow_golden_test.dart]: Making the shadows softer made the red color too subtle in the golden test, so I decided to switch the test to pitch black. Also, black shadows are the most commonly used ones.

File: lib/web_ui/lib/src/engine/canvas_pool.dart [hunk 2]
```diff
@@ -613,7 +613,12 @@ class _CanvasPool extends _SaveStackTracking {
...
         context.strokeStyle = '';
-        context.fillStyle = colorToCssString(color);
+        final int red = color.red;
+        final int green = color.green;
+        final int blue = color.blue;
+        // Multiply by 0.4 to make shadows less aggressive (https://github.com/flutter/flutter/issues/52734)
+        final int alpha = (0.4 * color.alpha).round();
+        context.fillStyle = colorComponentsToCssString(red, green, blue, alpha);
         context.shadowBlur = shadow.blurWidth;
         context.shadowColor = colorToCssString(color.withAlpha(0xff));
```
Comment (ditman) [inline lib/web_ui/lib/src/engine/canvas_pool.dart]: Why is this overriding the alpha channel of the shadow color to 0xff?

Can we do this instead?

```
context.shadowColor = colorToCssString(color);
```

(Assuming that color has the right alpha value set?)

Comment (yjbanov) [reply lib/web_ui/lib/src/engine/canvas_pool.dart]: @hterkelsen, do you recall why this was necessary? FWIW, the shadows look OK to me: https://github.com/flutter/goldens/blob/ae6003206eb721137c20cd56d8d1d8e2a76d6dd1/engine/web/shadows.png
